### PR TITLE
Issue #3736: Clean up import control configuration

### PR DESCRIPTION
**Pull request by jochenvdv**
Issue #3736

### CODE REVIEW

**Comment by romani** [comment]
@jochenvdv , please rebase your code over latest master it will fix wercker.

**Comment by rnveach** [comment]
@romani werker failure is because orekit is not pointing to commit where we did fixes for no whitespace before `...`.
https://github.com/checkstyle/checkstyle/blob/master/wercker.yml#L52

**Comment by romani** [comment]
Interesting how PR for '...' is passed

File: config/import-control.xml [hunk 1]
```diff
@@ -57,29 +63,33 @@
...
-    <allow class="com.puppycrawl.tools.checkstyle.Utils"
-           local-only="true"/>
+    <allow pkg="org.antlr.v4.runtime" local-only="true"/>
 
     <!-- allowed till https://github.com/checkstyle/checkstyle/issues/3455 -->
     <allow class="com.google.common.io.Closeables" local-only="true"/>
     <allow class="com.google.common.collect.ImmutableCollection" local-only="true"/>
     <allow class="com.google.common.collect.ImmutableMap" local-only="true"/>
+
+    <allow class="com.puppycrawl.tools.checkstyle.Checker"/>
```
Comment (romani) [inline config/import-control.xml]: looks like local-only should be used as much as possible , some times you do this, some time you do not.
please explain.

File: config/import-control.xml [hunk 2]
```diff
@@ -109,20 +119,20 @@
...
     <disallow pkg="com\.puppycrawl\.tools\.checkstyle\.checks\.[^.]+" regex="true"/>
-    <disallow pkg="com\.puppycrawl\.tools\.checkstyle\.(ant|doclets|gui)" regex="true"/>
+    <allow pkg="com.puppycrawl.tools.checkstyle.utils"/>
   </subpackage>
 
   <subpackage name="gui">
     <allow pkg="java.awt"/>
     <allow pkg="javax.swing"/>
-    <disallow pkg="com\.puppycrawl\.tools\.checkstyle\.(checks|ant|doclets|filters)" regex="true"/>
+    <allow pkg="com.puppycrawl.tools.checkstyle"/>
```
Comment (romani) [inline config/import-control.xml]: it sounds like alot ... can we restrict usage of other packages as much as possible ?

File: config/import-control.xml [hunk 3]
```diff
@@ -57,29 +63,33 @@
...
+    <allow class="com.puppycrawl.tools.checkstyle.Checker"/>
+    <!-- allowed till https://github.com/checkstyle/checkstyle/issues/3817 -->
+    <allow pkg="com.puppycrawl.tools.checkstyle.utils"/>
   </subpackage>
 
   <subpackage name="checks">
     <allow pkg="java.math"/>
-    <allow pkg="com.puppycrawl.tools.checkstyle.checks"/>
-    <allow class="com.puppycrawl.tools.checkstyle.Definitions"/>
+    <allow pkg="com.puppycrawl.tools.checkstyle"/>
```
Comment (romani) [inline config/import-control.xml]: it sounds like alot ... can we restrict usage of other packages as much as possible ?

File: config/import-control.xml (hunk 2 above)
Comment (jochenvdv) [inline config/import-control.xml]: Done.

File: config/import-control.xml (hunk 3 above)
Comment (jochenvdv) [inline config/import-control.xml]: Done.

File: config/import-control.xml (hunk 1 above)
Comment (jochenvdv) [inline config/import-control.xml]: I guess I forgot. Done.

**Comment by romani** [comment]
I am fine with changes, it is good step forward, if smth is missed - could be done later.

@rnveach , please review and merge.

**Pull request merged**